src/components/Inventory/AddInventory.js

```diff
@@ -0,0 +1,331 @@
+  }, []);
+
+  // This useEffect should run if no validation errors are found in the form inputs
+  useEffect(() => {
+    const addNewInventoryItem = async () => {
+      if (submissionData && validSubmissionCount > 0) {
+        try {
+          await axios.post(
+            "http://localhost:8080/api/inventories",
+            submissionData
+          );
+          alert("Form submitted successfully");
+          setSubmissionData(null);
+          setValidSubmissionCount(0);
+          setFormData(initialState.current);
+        } catch (err) {
+          console.error("Failed to submit form:", err);
+        }
+      }
+    };
+    addNewInventoryItem();
+  }, [submissionData, validSubmissionCount, initialState]);
+
+  return (
+    <>
+      {warehousesLoaded ? (
+        <div className="card-container">
+          <h2 className="card-container__heading">
+            <img
+              src={BackIcon}
+              alt="Back Icon"
+              className="inventory-card__icon"
+              onClick={() => {
+                navigate(`/inventory`);
+              }}
+            />
+            Add New Inventory Item
+          </h2>
+          <hr className="inventory-card__divider"></hr>
+
+          <form className="form">
+            <section className="form--left">
+              <h2 className="form-heading">Item Details</h2>
+
+              {/* name Input */}
+              <label className="form-label">Item Name</label>
+              <input
+                type="text"
+                name="item_name"
+                value={formData.item_name}
+                onChange={handleChange}
+                placeholder="Item Name"
+                className={`form-input ${
+                  errors.item_name ? "form-input--error" : ""
+                }`}
+              />
+              {errors.item_name ? (
+                <p className="form-error">
+                  <img
+                    src={ErrorIcon}
+                    alt="Error Icon"
+                    className="form-error__icon"
+                  />
+                  This field is required
+                </p>
+              ) : null}
+              <br></br>
+
+              {/* Description Input */}
+              <label className="form-label">Description</label>
+              <textarea
+                name="description"
+                value={formData.description}
+                onChange={handleChange}
+                placeholder="Please enter a brief item description..."
+                className={`form-input--description ${
+                  errors.description ? "form-input--error" : ""
+                }`}
+              />
+              {errors.description ? (
+                <p className="form-error">
+                  <img
+                    src={ErrorIcon}
+                    alt="Error Icon"
+                    className="form-error__icon"
+                  />
+                  This field is required
+                </p>
```

Comment:
This JSX is repeating quite a bit. 

Would be good to move it to another function and call that function here